From 6d1fbe5b3d7231fed2eef78d4d8ea3c2c6e80269 Mon Sep 17 00:00:00 2001
From: Algea Cao <algea.cao@rock-chips.com>
Date: Wed, 4 Nov 2020 16:48:59 +0800
Subject: [PATCH] drm: rockchip: rk628: post_process: Enable y2r when input
 mode clk is 594Mhz

Because hdmirx phy only support yuv420 when input mode is 4K-60Hz,
y2r should be enabled.

Change-Id: I1eca2c6783c63e07cb13b617bb32ff81f2229c7d
Signed-off-by: Algea Cao <algea.cao@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rk628/rk628_post_process.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/rockchip/rk628/rk628_post_process.c b/drivers/gpu/drm/rockchip/rk628/rk628_post_process.c
index af9aca69a267..4e2a5d56333d 100644
--- a/drivers/gpu/drm/rockchip/rk628/rk628_post_process.c
+++ b/drivers/gpu/drm/rockchip/rk628/rk628_post_process.c
@@ -275,6 +275,10 @@ static void rk628_post_process_bridge_mode_set(struct drm_bridge *bridge,
 		drm_mode_copy(&pp->dst_mode, &rk628->dst_mode);
 	else
 		drm_mode_copy(&pp->dst_mode, &pp->src_mode);
+
+	/* hdmirx 4k-60Hz mode only support yuv420 */
+	if (pp->src_mode.clock == 594000)
+		regmap_write(pp->grf, GRF_CSC_CTRL_CON, SW_Y2R_EN(1));
 }
 
 static int rk628_post_process_bridge_attach(struct drm_bridge *bridge)
-- 
2.35.3

